backend/frontend/tests/test_course_component_details_pages.py

```diff
@@ -69,7 +69,58 @@ def test_link_project_page_displays_correct_details(self):
         self.assertIn("Feb. 13, 2024, 2:06 p.m.", body)
         self.assertIn("learner_reviewer@umuzi.org", body)
         self.assertIn(
-            "https://raw.githubusercontent.com/Umuzi-org/tech-department/master/content/projects/tdd/simple-calculator/part-1/_index.md",
+            self.recruit_project.content_url,
             body,
         )
         self.assertIn("No link submitted yet", body)
+
+    def test_link_submission_form_correctly_updates_link_submission_after_link_update(
+        self,
+    ):
+        self.make_ip_project_card(ContentItem.LINK)
+
+        self.link_project_url = self.reverse_url(
+            "course_component_details", kwargs={"project_id": self.recruit_project.id}
+        )
+        self.page.goto(self.link_project_url)
+
+        body = self.page.text_content("body")
```

Comment:
You may also want to preemptively wait for everything to load before selecting any text_contethis tends to cause failing tests if something is selected while things haven't fully loaded. You can look up examples in other tests or maybe there's a better way